src/Microsoft.HttpRepl/VerbosityLogger.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using Microsoft.Repl.ConsoleHandling;
+
+namespace Microsoft.HttpRepl
+{
+    internal class VerbosityLogger
+    {
+        private readonly bool _isVerbosityEnabled;
+        private readonly IConsoleManager? _consoleManager;
+
+        private VerbosityLogger(IConsoleManager? consoleManager, bool isVerbosityEnabled)
+        {
+            _consoleManager = consoleManager;
+            _isVerbosityEnabled = isVerbosityEnabled;
+        }
+
+        public static VerbosityLogger FromConsoleManager(IConsoleManager? consoleManager, bool isVerbosityEnabled = false)
```

Comment:
What does a static factory method get you that a public constructor wouldn't, when they take the same args?  It's a more confusing API to have to discover that I can't just new() it up.